docs/reference/content/driver-reactive/getting-started/quick-start.md

```diff
@@ -0,0 +1,455 @@
+                .append("info", new Document("x", 203).append("y", 102));
+ ```
+
+{{% note %}}
+The BSON type of array corresponds to the Java type `java.util.List`. For a list of the BSON type and the corresponding type in Java, see the [Document]({{< relref "bson/documents.md#document" >}}) mapping.
+{{% /note %}}
+
+
+## Insert a Document
+
+Once you have the `MongoCollection` object, you can insert documents into the collection.
+
+### Insert One Document
+
+To insert a single document into the collection, you can use the collection's [`insertOne()`]({{< apiref "com/mongodb/reactivestreams/client/MongoCollection.html#insertOne-TDocument-" >}}) method.
+
+```java
+collection.insertOne(doc).subscribe(new OperationSubscriber<Success>());
+```
+
+{{% note %}}
+If no top-level `_id` field is specified in the document, MongoDB automatically adds the `_id` field to the inserted document.
+{{% /note %}}
+
+{{% note class="important" %}}
+In the API all methods returning a `Publisher` are "cold" streams meaning that nothing happens until they are Subscribed to.
+
+The example below does nothing:
+
+```java
+Publisher<Success> publisher = collection.insertOne(doc);
+```
+
+Only when a `Publisher` is subscribed to and data requested will the operation happen:
+
+```java
+publisher.subscribe(new Subscriber<Success>() {
+    @Override
+    public void onSubscribe(final Subscription s) {
+        s.request(1);  // <--- Data requested and the insertion will now occur
+    }
+
+    @Override
+    public void onNext(final Success success) {
+        System.out.println("Inserted");
+    }
+
+    @Override
+    public void onError(final Throwable t) {
+        System.out.println("Failed");
+    }
+
+    @Override
+    public void onComplete() {
+        System.out.println("Completed");
+    }
+});
+```
+
+Once the document has been inserted the `onNext` method will be called and it will
+print "Inserted!" followed by the `onComplete` method which will print "Completed".  
+If there was an error for any reason the `onError` method would print "Failed".
+
+{{% /note %}}
+
+### Insert Multiple Documents
+
+To add multiple documents, you can use the collection's [`insertMany()`]({{< apiref "com/mongodb/reactivestreams/client/MongoCollection.html#insertMany-java.util.List-" >}}) method which takes a list of documents to insert.
+
+The following example will add multiple documents of the form:
+
+```javascript
+{ "i" : value }
+```
+
+Create the documents in a loop and add to the `documents` list:
+
+
+```java
+List<Document> documents = new ArrayList<Document>();
+for (int i = 0; i < 100; i++) {
+    documents.add(new Document("i", i));
+}
+```
+
+To insert these documents to the collection, pass the list of documents to the
+[`insertMany()`]({{< apiref "com/mongodb/reactivestreams/client/MongoCollection.html#insertMany-java.util.List-" >}}) method.
+
+```java
+subscriber = new ObservableSubscriber<Success>();
```

Comment:
The reason for the different subscribers and blocking is because that is how the code is written in the QuickTour.java.

We wait for the insert to happen before counting / finding data.